File: Backend/TrackrAPI/TrackrAPI/Services/Seguridad/ExpedientePadecimientoService.cs
using TrackrAPI.Dtos.Seguridad;
using TrackrAPI.Models;
using TrackrAPI.Repositorys.Seguridad;

namespace TrackrAPI.Services.Seguridad
{
    public class ExpedientePadecimientoService
    {
        private IExpedientePadecimientoRepository expedientePadecimientoRepository;

        public ExpedientePadecimientoService(
            IExpedientePadecimientoRepository expedientePadecimientoRepository
            )
        {
            this.expedientePadecimientoRepository = expedientePadecimientoRepository;

        }

        public IEnumerable<ExpedientePadecimientoSelectorDTO> ConsultarParaSelector()
        {
            return expedientePadecimientoRepository.ConsultarParaSelector();
        }

        public IEnumerable<ExpedientePadecimientoDTO> ConsultarPorUsuario(int idUsuario)
        {
            return expedientePadecimientoRepository.ConsultarPorUsuario(idUsuario);
        }

    }
}

File: Frontend/trackr-administrador/src/app/shared/http/seguridad/expediente-padecimiento.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ExpedientePadecimientoDTO } from '@dtos/seguridad/expediente-padecimiento-dto';
import { ExpedientePadecimientoSelectorDTO } from '@dtos/seguridad/expediente-padecimiento-selector-dto';

@Injectable({
    providedIn: 'root'
})
export class ExpedientePadecimientoService {
    private dataUrl = 'expedientePadecimiento/';

    constructor(public http: HttpClient) {}

    public consultarPorUsuario(idUsuario: number) {
        return this.http.get<ExpedientePadecimientoDTO>(this.dataUrl + `consultarPorUsuario/${idUsuario}`);
    }

    public consultarParaSelector(){
        return this.http.get<ExpedientePadecimientoSelectorDTO[]>(this.dataUrl + `consultarParaSelector`);
    }

}